README.md

```diff
@@ -77,13 +77,11 @@ If you use the common trusted custodians for all users, you can deploy a new DKI
 If each user should be able to modify the registered public keys, a new DKIM registry contract needs to be deployed for each user.
 
 The email-auth contract in `EmailAuth.sol` is a contract for each email user.
-Its contract Ethereum address is calculated as the CREATE2 of the account salt, i.e., the hash of the user's email address and one account code held by the user. 
-It provides an entry function `authEmail` to verify the email-auth message by calling the verifier and the DKIM registry contracts.
-<!-- Besides, it supports [ERC-1271](https://eips.ethereum.org/EIPS/eip-1271).  -->
-After the email-auth message is processed in the `authEmail` function, the `isValidSignature` function of the email-auth contract returns true for the hash of the data in the email-auth message and its email nullifier, a 32-byte data unique to each email.
+Its contract Ethereum address is derived from its initial owner address, an address of a controller contract that can define the supported subject templates and the account salt, i.e., the hash of the user's email address and one account code held by the user, through CREATE2.
+It provides a function `authEmail` to verify the email-auth message by calling the verifier and the DKIM registry contracts.
 
 Your application contract can employ those contracts in the following manner:
-1. For a new email user, the application contract deploys (a proxy of) the email-auth contract. Subsequently, the application contract sets the addresses of the verifier and the DKIM registry contracts and some subject templates for your application to the email-auth contract.
+1. For a new email user, the application contract deploys (a proxy of) the email-auth contract. Subsequently, the application contract sets the addresses of the verifier and the DKIM registry contracts and some subject templates for your application to the email-auth contract as its controller contract.
```

Comment:
> for your application to the email-auth contract as its controller contract

The "as its controller contract" part of the sentence doesn't make sense to me. Would something like this make sense:

```suggestion
1. For a new email user, the application contract deploys (a proxy of) the email-auth contract. Subsequently, the application contract sets the addresses of the verifier and the DKIM registry contracts and some subject templates for your application to the email-auth contract. The controller contract that has permissions to modify `EmailAuth` instances is also set.
```

Reply:
Thank you for your suggestion!
In that context, the application contract is the same as the controller contract.
How about the following modification?
```
Subsequently, the application contract sets the addresses of the verifier and the DKIM registry contracts and some subject templates for your application to the email-auth contract. Here, the email-auth contract registers the application contract as a controller contract that has permissions to modify the subject templates. 
```

Reply:
Got it. Yeah I like your suggestion